### PR TITLE
Dp 6845 header

### PR DESCRIPTION
**Pull request by aditudorache**
## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [x] The default export from a file matches the file name
- [x] The component styles are only placed in the `asc-core/src/components` folder
- [x] The component style name follows the pattern `<ComponentName>Style/<ComponentName>Style.ts`
- [x] Pull request has tests (we are going for 100% coverage!)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-ui/src/components/Header/Header.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react'
+import { storiesOf } from '@storybook/react'
+import { ThemeProvider } from '../../index'
+import Header from './Header'
+import ContentFiller from '../../internals/ContentFiller/ContentFiller'
+
+const SimpleComponent: React.FC<{}> = () => (
+  <ThemeProvider themeName="default">
+    <React.Fragment>
```
Comment (royderks) [inline packages/asc-ui/src/components/Header/Header.stories.tsx]: Use shorthand notation <>

File: packages/asc-core/src/theme/index.ts [hunk 2]
```diff
@@ -15,6 +15,23 @@ export namespace Theme {
 
   export type GlobalStyleType = string
 
+  export type TypeBreakpoint = 'min-width' | 'max-width'
+
+  export interface GetBreakpointFunc {
```
Comment (royderks) [inline packages/asc-core/src/theme/index.ts]: Why not just GetBreakpoint?

File: packages/asc-ui/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    position: fixed;
+    top: 0;
+  }
+`
+
+const Header: React.FC<Props> = ({ ...otherProps }) => (
+  <HeaderGridWrapper>
+    <HeaderGridContainer>
+      <HeaderStyleLocal {...otherProps}>
+        <HeaderTitle title="Data en informatie" />
```
Comment (royderks) [inline packages/asc-ui/src/components/Header/Header.tsx]: Make this a variable so other titles can be used as well

File: packages/asc-ui/src/components/Header/HeaderLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import styled from '../../styled-components'
+
+type Props = {
+  homeLink?: string
+}
+
+const HeaderLogoStyle = styled.a<{}>`
```
Comment (royderks) [inline packages/asc-ui/src/components/Header/HeaderLogo.tsx]: Why styling in the asc-ui and not asc-core?

File: packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  & > div {
+    align-self: center;
+    flex-grow: 1;
+    text-align: center;
+  }
+`
+
+const CenteredElement: React.FC<{}> = ({ children }) => (
+  <CenteredElementStyle>
+    <div>{children}</div>
```
Comment (royderks) [inline packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx]: Could this also be done without rendering another element to the DOM?

File: packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react'
+import styled from '../../styled-components'
+
+const CenteredElementStyle = styled.div`
```
Comment (royderks) [inline packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx]: Why styling in asc-ui?

File: packages/asc-ui/src/internals/ContentFiller/ContentFiller.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import styled from '../../styled-components'
+import Typography from '../../components/Typography'
+
+type ContentFillerStyleProps = {
+  backgroundColor?: string
+  marginTop?: string
+  marginTopBorder?: string
+}
+
+const ContentFillerStyle = styled.div<ContentFillerStyleProps>`
```
Comment (royderks) [inline packages/asc-ui/src/internals/ContentFiller/ContentFiller.tsx]: See previous comments about styling in asc-ui

File: packages/asc-ui/src/components/Header/HeaderTitle.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    line-height: 1.5625rem;
+    padding-top: 6px;
+  }
+
+  & > h1 > span > a {
+    color: #000;
+    text-decoration: none;
+    line-height: 2.1875rem;
+
+    @media screen and (max-width: 360px) {
```
Comment (royderks) [inline packages/asc-ui/src/components/Header/HeaderTitle.tsx]: Get breakpoints from theme

File: packages/asc-ui/src/components/Header/HeaderTitle.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import HeaderLogo from './HeaderLogo'
+import styled from '../../styled-components'
+import { getBreakpointFromTheme } from '../../styles/utils'
+
+type Props = {
+  title: string
+}
+
+const HeaderTitleStyle = styled.div<{}>`
```
Comment (royderks) [inline packages/asc-ui/src/components/Header/HeaderTitle.tsx]: Move to asc-core

File: packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx (hunk 5 above)
Comment (aditudorache) [reply packages/asc-ui/src/internals/CenteredElement/CenteredElement.tsx]: no

File: packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts [hunk 10]
```diff
@@ -25,7 +25,7 @@ const ButtonStyle = styled(ButtonBaseStyle)<Props>`
   & svg {
     width: 30px;
```
Comment (royderks) [inline packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts]: Is 30px the standard for svg's in this theme? If so, can we add this to the theme config?

File: packages/asc-core/src/components/HeaderStyle/HeaderLogoStyle.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    font-weight: 700;
+    height: 0;
+    position: absolute;
+    width: 0;
+  }
+
+  min-height: 50px;
+
+  & > svg {
+    height: 30px;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderLogoStyle.ts]: Standard for svg?

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  & > h1 > span {
+    display: inline-block;
+    line-height: 1.5625rem;
+    padding-top: 6px;
+  }
+
+  & > h1 > span > a {
+    color: #000;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: Get this from the theme?

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    text-decoration: none;
+    line-height: 2.1875rem;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: Where is this value based on, is it a default?

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  & > h1 {
+    display: flex;
+    height: 100%
+    margin: 0;
+    background-color: rgba(192, 192, 192, 0.1);
+  }
+
+  & > h1 > span {
+    display: inline-block;
+    line-height: 1.5625rem;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: Same as below

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  & > h1 > span > a {
+    color: #000;
+    text-decoration: none;
+    line-height: 2.1875rem;
+
+    @media screen and ${({ theme }) =>
+      getBreakpointFromTheme(theme, 'max-width', 'mobileM')} {
+      line-height: 18px;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: Why fixed here and not rem from above?

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+      font-size: 14px;
+    }
+
+    @media screen and ${({ theme }) =>
+      getBreakpointFromTheme(theme, 'min-width', 'mobileM')} {
+      line-height: 20px;
```
Comment (royderks) [inline packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: Same

File: packages/asc-ui/src/internals/Box/BoxWrapper.tsx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+
+export type Props = {}
+
+export const BoxWrapperStyle = styled.div`
+  position: relative;
+  margin: 0;
+  padding: 0;
+  border: 0;
+  vertical-align: baseline;
+  font-family: 'AvenirNextLTW01-Regular', arial, sans-serif;
```
Comment (royderks) [inline packages/asc-ui/src/internals/Box/BoxWrapper.tsx]: Get font from theme?

File: packages/asc-ui/src/internals/Box/BoxWrapper.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react'
+import styled from '../../styled-components'
+
+export type Props = {}
+
+export const BoxWrapperStyle = styled.div`
```
Comment (royderks) [inline packages/asc-ui/src/internals/Box/BoxWrapper.tsx]: Are internals only used by a specific UI? Or should they be globally available to other UIs in the future as well?

File: packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts (hunk 10 above)
Comment (aditudorache) [reply packages/asc-core/src/components/ButtonStyle/ButtonStyle.ts]: is fixed width, no need for config

File: packages/asc-core/src/components/HeaderStyle/HeaderLogoStyle.ts (hunk 11 above)
Comment (aditudorache) [reply packages/asc-core/src/components/HeaderStyle/HeaderLogoStyle.ts]: just specific for this one

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts (hunk 12 above)
Comment (aditudorache) [reply packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: is a constant, not need for a theme

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts (hunk 13 above)
Comment (aditudorache) [reply packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: removed

File: packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts (hunk 14 above)
Comment (aditudorache) [reply packages/asc-core/src/components/HeaderStyle/HeaderTitleStyle.ts]: removed

File: packages/asc-ui/src/internals/Box/BoxWrapper.tsx (hunk 17 above)
Comment (aditudorache) [reply packages/asc-ui/src/internals/Box/BoxWrapper.tsx]: not needed, added extra info in the pull request.md

File: packages/asc-ui/src/internals/Box/BoxWrapper.tsx (hunk 18 above)
Comment (aditudorache) [reply packages/asc-ui/src/internals/Box/BoxWrapper.tsx]: just internal used

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 19]
```diff
@@ -3,7 +3,7 @@
 Before opening a pull request, please ensure:
 
 - [ ] The default export from a file matches the file name
-- [ ] The component styles are only placed in the `asc-core/src/components` folder
+- [ ] The component styles are only placed in the `asc-core/src/components` folder. Exception are the `asc-core/src/internals` folder where the styles allowed. The components from this folder are just for internal use in the stories.
```
Comment (royderks) [inline .github/PULL_REQUEST_TEMPLATE.md]: folder where the styles _are_ allowed

File: README.md [hunk 20]
```diff
@@ -69,6 +69,14 @@ To test your components in other repo's, do the following:
 
 Now you can import the package like you would do like a normal npm dependency. Changes you will make in your package will be seen in your repo.
 
+#### Development guidelines
+
+- The default export from a file matches the file name
+- The component styles are only placed in the `asc-core/src/components` folder.
+  Exception are the `asc-core/src/internals` folder where the styles allowed. The components from this folder are just for internal use in the stories.
```
Comment (royderks) [inline README.md]: folder where the styles _are_ allowed

**Pull request merged**